lib/transmuxer/ts_transmuxer.js

```diff
@@ -144,6 +149,29 @@ shaka.transmuxer.TsTransmuxer = class {
   }
 
 
+  /**
+   * @param {string} codec
+   * @return {string}
+   * @private
+   */
+  getCorrectVideoCodecs_(codec) {
+    if (codec.includes('avc1')) {
+      // Convert avc1 codec string from RFC-4281 to RFC-6381 for
+      // MediaSource.isTypeSupported
+      // Example, convert avc1.66.30 to avc1.42001e (0x42 == 66 and 0x1e == 30)
+      const avcdata = codec.split('.');
+      if (avcdata.length == 3) {
+        let result = avcdata.shift() + '.';
+        result += parseInt(avcdata.shift(), 10).toString(16);
+        result +=
```

Comment:
You could use the `padStart` string method for this. Like...
`result += parseInt(avcdata.shift(), 10).toString(16).padStart(4, "0")`
Technically doing this would cause weird things to happen if the level is `0x10000` or above, but according to Wikipedia the highest current level is `0x3E` so a five byte level string is... pretty far away.
And I think that using `padStart` would make the intent of the line a little clearer.

Reply:
After some discussion in chat, Alvaro pointed out that `padStart` is not supported by Tizen. Drat!

Reply:
`padStart` is not supported in Tizen 3 for example, do you know if closure polyfill it?

See: https://caniuse.com/pad-start-end